src/ripple/ledger/impl/CashDiff.cpp

```diff
@@ -0,0 +1,731 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/ledger/CashDiff.h>
+#include <ripple/ledger/detail/ApplyStateTable.h>
+#include <ripple/protocol/st.h>
+#include <boost/container/static_vector.hpp>
+#include <cassert>
+#include <cstdlib>                                  // std::abs()
+
+namespace ripple {
+namespace detail {
+
+// Data structure that summarize cash changes in a single ApplyStateTable.
+struct CashSummary
+{
+    // Sorted vectors.  All of the vectors fill in for std::maps.
+    std::vector<std::pair<
+        AccountID, XRPAmount>> xrpChanges;
+
+    std::vector<std::pair<
+        std::tuple<AccountID, AccountID, Currency>, STAmount>> trustChanges;
```

Comment:
A pair where the first element is a 2-element tuple seems a little strange. I'd probably flatten these (tho no change needed)